src/components/dashboards/index.tsx

```diff
@@ -613,6 +617,99 @@ const Card: React.FC<CardProps> = React.memo(({ panel }) => {
   const [extensions, extensionsResolved] = useResolvedExtensions<DataSourceExtension>(isDataSource);
   const hasExtensions = !_.isEmpty(extensions);
 
+  // const [isOpen, setIsOpen] = React.useState(false);
+
+  // const onToggle = (isOpen: boolean) => {
+  //   setIsOpen(isOpen);
+  // };
+
+  // const onFocus = () => {
+  //   const element = document.getElementById('toggle-kebab');
+  //   element.focus();
+  // };
+
+  // const onSelect = () => {
+  //   setIsOpen(false);
+  //   onFocus();
+  // };
+  const formatSeriesTitle = React.useCallback(
+    (labels, i) => {
+      const title = panel.targets?.[i]?.legendFormat;
+      if (_.isNil(title)) {
+        return _.isEmpty(labels) ? '{}' : '';
+      }
+      // Replace Prometheus labels surrounded by {{ }} in the graph legend label templates
+      // Regex is based on https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+      // with additional matchers to allow leading and trailing whitespace
+      return title.replace(
+        /{{\s*([a-zA-Z_][a-zA-Z0-9_]*)\s*}}/g,
+        (match, key) => labels[key] ?? '',
+      );
+    },
+    [panel],
+  );
+
+  const [csvData, setCsvData] = React.useState([]);
+
+  const csvExportHandler = () => {
+    let csvString = '';
+    const result = {};
+    const nameArray = []; //
+
+    for (const [i, row] of csvData.entries()) {
+      for (const item of row) {
+        for (const entry of item[1]) {
+          const dateTime = entry.x.toISOString();
+          const value = entry.y;
+          if (!result[dateTime]) {
+            result[dateTime] = [];
+          }
+          const name = formatSeriesTitle(item[0], 0);
+          nameArray.push(name); // once per row?
+          if (!name) {
+            continue;
+          }
+          //const series = item[0][name].toString()
+          result[dateTime].push({ [name]: value });
+        }
+      }
+    }
+    const firstDateTime = Object.keys(result)[0];
+
+    const uniqueVerbs = new Set();
```

Comment:
This only applies for one chart example, would it be better to have?
```suggestion
    const uniqueSeries = new Set();
```